packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -640,7 +644,7 @@ export class GarbageCollector implements IGarbageCollector {
 
         // Get the GC data from the base snapshot. Use LazyPromise because we only want to do this once since it
         // it involves fetching blobs from storage which is expensive.
-        const baseSnapshotDataP = new LazyPromise<IGCSnapshotData | undefined>(async () => {
+        this.baseSnapshotDataP = new LazyPromise<IGCSnapshotData | undefined>(async () => {
```

Comment:
Now that I think about it, this isn't really a lazy promise anymore is it because it's almost executed immediately on load

Reply:
It is called at multiple places so having it as lazy promise ensures its only executed once.